File: scripts/R/lr_age_sales.r
# Load required libraries
library(ggplot2)
library(dplyr)

# Assuming 'dataset' is your Power BI dataset

# Scatter plot
ggplot(dataset, aes(x = Age, y = Sales)) +
  geom_point(color = '#0D6ABF', alpha = 0.6) +
  labs(x = 'Age', y = 'Sales', title = 'Scatter Plot Age vs. Sales') +
  theme_minimal()

# Fit a linear regression model
model <- lm(Sales ~ Age, data = dataset)

# Create data for the regression line
age_range <- seq(min(dataset$Age), max(dataset$Age), length.out = 100)
predicted_sales <- predict(model, newdata = data.frame(Age = age_range))

# Add the regression line to the plot
ggplot(dataset, aes(x = Age, y = Sales)) +
  geom_point(color = '#0D6ABF', alpha = 0.6) +
  geom_line(data = data.frame(Age = age_range, Sales = predicted_sales), 
            aes(x = Age, y = Sales), color = 'red', size = 1) +
  labs(x = 'Age', y = 'Sales', title = 'Scatter Plot Age vs. Sales') +
  theme_minimal()

# Display the plot
print(last_plot())

File: scripts/Python/lr_age_sales.py
# The following code to create a dataframe and remove duplicated rows is always executed and acts as a preamble for your script: 

# Paste or type your script code here:
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.linear_model import LinearRegression

# dataset is the DataFrame provided by Power BI
# Scatter plot
plt.figure(figsize=(10, 6))
plt.scatter(dataset['Age'], dataset['Sales'], color='#0D6ABF', label='Data points')

# Fit a linear regression model
model = LinearRegression()
X = dataset[['Age']]  # Independent variable
y = dataset['Sales']  # Dependent variable
model.fit(X, y)

# Predict values for the line
X_fit = pd.DataFrame({'Age': range(int(dataset['Age'].min()), int(dataset['Age'].max()) + 1)})
y_fit = model.predict(X_fit)

# Plot the regression line
plt.plot(X_fit, y_fit, color='red', linewidth=2, label='Fit Line')

# Add labels and title
plt.xlabel('Age')
plt.ylabel('Sales')
plt.title('Scatter Plot Age vs. Sales')
plt.legend()

# Show plot
plt.show()


